src/Sarif.Viewer.VisualStudio/ErrorList/SarifTableDataSource.cs

```diff
@@ -4,69 +4,34 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.ComponentModel.Composition;
 using System.Linq;
 using System.Threading;
 
-using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
-using Microsoft.VisualStudio.Shell.TableControl;
 using Microsoft.VisualStudio.Shell.TableManager;
 using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.Sarif.Viewer.ErrorList
 {
-    internal class SarifTableDataSource : ITableDataSource, IDisposable
+    internal class SarifTableDataSource : SarifTableDataSourceBase, IDisposable
     {
         private static SarifTableDataSource _instance;
         private readonly ReaderWriterLockSlimWrapper sinksLock = new ReaderWriterLockSlimWrapper(new ReaderWriterLockSlim());
-        private readonly List<SarifTableDataSink> sinks = new List<SarifTableDataSink>();
+        private readonly List<SinkHolder> sinkHolders = new List<SinkHolder>();
```

Comment:
>new List<SinkHolder>(); [](start = 56, length = 23)

NIT: Just a style thing, please move all inlined field initializers to the private ctor. The reason we do this is to gather all non-default init in a common place and to prevent the debugger from jumping around unduly when stepping through.